python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -568,7 +662,30 @@ def on_task_finished(self, task_index: int, exception: Optional[Exception]):
                     assert meta.size_bytes is not None
                     self.obj_store_mem_spilled += meta.size_bytes
 
+                    if self._per_node_metrics_enabled:
+                        if meta.exec_stats.node_id is not None:
+                            node_metrics = self._per_node_metrics[
+                                meta.exec_stats.node_id
+                            ]
+                        else:
+                            node_metrics = self._per_node_metrics[_NODE_UNKNOWN]
+                        node_metrics.obj_store_mem_spilled += meta.size_bytes
+
         self.obj_store_mem_freed += total_input_size
 
+        # update per node metrics
+        if self._per_node_metrics_enabled:
+            node_ids_seen = set()
+            for meta, node_metrics, node_id in iter_meta_and_node_metrics(
+                inputs, self._per_node_metrics
+            ):
+                # stats to update once per node id or if node id is unknown
+                if node_id not in node_ids_seen or node_id == _NODE_UNKNOWN:
+                    node_metrics.num_tasks_finished += 1
+                    node_metrics.num_tasks_running -= 1
```

Comment:
In a similar vein as my comment about `num_tasks_running`, I'm not really sure how I'd interpret `num_tasks_finished`. If I'm understanding correctly, it represents the number of tasks finished that have an input from this node, rather than the number of finished tasks that ran on this node.